pkg/gardener/gardener_error_handler.go

```diff
@@ -0,0 +1,45 @@
+package gardener
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	gardener "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardenerhelper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+)
+
+type ErrReason string
+
+func IsRetryable(lastErrors []gardener.LastError) bool {
+	if len(lastErrors) > 0 &&
+		!gardenerhelper.HasNonRetryableErrorCode(lastErrors...) {
+		return true
+	}
+	return false
+}
+
+func ToErrReason(lastErrors ...gardener.LastError) ErrReason {
+	var codes []gardener.ErrorCode
+	var vals []string
+
+	for _, e := range lastErrors {
+		if len(e.Codes) > 0 {
+			codes = append(codes, e.Codes...)
+		}
+	}
+
+	for _, code := range codes {
+		vals = append(vals, string(code))
+	}
+
+	return ErrReason(strings.Join(vals, ", "))
+}
+
+func CombineErrorDescriptions(lastErrors []gardener.LastError) string {
```

Comment:
not yet used as I've created it without knowing that the ToErrReason existed in a different place. Kept it as it combined more human readable description